src/Components/Web.Extensions/src/HeadManagement/HeadTagBase.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    /// <summary>
+    /// Serves as a base for components that represent tags in the HTML head.
+    /// </summary>
+    public abstract class HeadTagBase : ComponentBase, IDisposable
+    {
+        private readonly string _id = Guid.NewGuid().ToString("N");
+
+        private TagElement _tagElement;
+
+        private HeadManager _headManager = default!;
+
+        [Inject]
+        private IServiceProvider ServiceProvider { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets a collection of additional attributes that will be applied to the meta element.
+        /// </summary>
+        [Parameter(CaptureUnmatchedValues = true)]
+        public IReadOnlyDictionary<string, object>? Attributes { get; set; }
+
+        /// <summary>
+        /// Gets the name of the tag being represented.
+        /// </summary>
+        protected abstract string TagName { get; }
+
+        /// <inheritdoc />
+        protected override void OnInitialized()
+        {
+            _headManager = ServiceProvider.GetHeadManager() ??
+                throw new InvalidOperationException($"{GetType()} requires a {typeof(HeadManager)} service.");
```

Comment:
@SteveSandersonMS \ @javiercn, would we consider having a way for `Inject` to tell you what to do to fix this?

```C#
[Inject(Error = "Did you forget to invoke AddBlah()?")]
```

It shouldn't be too much work to include that in the diagnostics here: https://github.com/dotnet/aspnetcore/blob/0ec15ea01d20597d1c8cea7f8df51d26a07d5f87/src/Components/Components/src/ComponentFactory.cs#L80